### PR TITLE
refactor(semconv): Add more unit tests, fix inconsistencies

### PR DESCRIPTION
**Pull request by lquerel**
This PR has the following goals:
- Prepare `weaver_semconv` public API for its first release
- Test all error paths and increase test coverage
- Remove all experimental codes
- Make this crate easy to use and contribute to.

List of updates:
- [X] Improve the `weaver_semconv` method signatures.
- [X] Improve the `weaver_semconv` documentation. 
- [X] Reorganize Rust files in `weaver_semconv` to maintain a single purpose per file.
- [X] Remove all unused structs, enums, and methods in `weaver_semconv`.
- [X] Achieve 80% or higher test coverage and test all error paths in `weaver_semconv`.
- [X] Simplify dependency graph between internal crate (see docs/images/dependencies.svg)
- [X] Remove dependency on the validator crate.
- [X] Remove dependency on terra (replaced by MiniJinja).
- [X] Remove all code related to the Application Telemetry Schema and corresponding CLI commands (i.e. experimental commands).

Open questions for discussion:
- Should we discontinue support for GroupType::Scope within the weaver_semconv crate?
- Should we remove support for constraints within the weaver_semconv crate?

Important note: There is still an important refactor before to consider this crate ready for a first release. This refactoring is described in https://github.com/open-telemetry/weaver/issues/124. This modification will be included in a separate PR.

Some of the removed code will potentially be reintroduced once the app telemetry schema spec will be finalized.

### CODE REVIEW

File: crates/weaver_semconv/src/lib.rs [hunk 1]
```diff
@@ -1,65 +1,49 @@
...
 
-use std::collections::{HashMap, HashSet};
-use std::fs::File;
-use std::io::BufReader;
-use std::path::Path;
-
-use glob::glob;
-use serde::{Deserialize, Serialize};
-use validator::Validate;
+use std::collections::HashMap;
```
[the remark this answers is not in the document]
Comment (lquerel) [reply crates/weaver_semconv/src/lib.rs]: Fixed

[the remark this answers is not in the document]
Comment (lquerel) [reply crates/weaver_semconv/src/lib.rs]: Fixed

File: crates/weaver_checker/src/lib.rs [hunk 2]
```diff
@@ -76,6 +83,16 @@ impl Error {
     }
 }
 
+/// Handles a list of errors and returns a compound error if the list is not
+/// empty or () if the list is empty.
+pub fn handle_errors(errors: Vec<Error>) -> Result<(), Error> {
```
Comment (jsuereth) [inline crates/weaver_checker/src/lib.rs:88]: I have an idea for  how to make this method a bit more generic. 

TL;DR; I think we could use a `trait`  on specific `Error` types, with a macro.

So you could do something like:

enum Error {
 ...,
 
 #[compound_error]
 CompoundError(errors: Vec<Error>)
}
```

And then we'd have a method like:

```
fn handle_errors<E: WeaverError>(errors:  Vec<E>) -> Result<(), E>
```

**Review by jsuereth — APPROVED** [review]
Lots of things to review here.  Overall LGTM.

File: crates/weaver_checker/src/lib.rs (hunk 2 above)
Comment (lquerel) [reply crates/weaver_checker/src/lib.rs:88]: Yes interesting.
